book/src/SUMMARY.md

```diff
@@ -1,9 +1,8 @@
 # Summary
 
-- Components
```

Comment:
I think a Components Section would be better. We could add some small documentation about the other components like Catalogue, Backends etc. as well

Reply:
I imagined a split between "User guide" for the users of lens and "Development" for the developers of lens. But yeah it might be sense to make a components section under "user guide"? Or change the structure entirely, I'm not too set on anything. I mostly made this change because SUMMARY.md was just broken before and not building the book. We should probably add book building to CI so we see if the book still builds on every PR.